sherpa/plot/__init__.py

```diff
@@ -73,15 +73,19 @@
 _ = numpy.seterr(invalid='ignore')
 
 plot_opt = config.get('options', 'plot_pkg', fallback='dummy')
-plot_opt = [o.strip().lower() + '_backend' for o in plot_opt.split()]
+plot_opt = [o.strip().lower() for o in plot_opt.split()]
+
+plot_backends = {'pylab': 'sherpa.plot.pylab_backend.PylabBackend',
+                 'matplotlib': 'sherpa.plot.pylab_backend.PylabBackend',
+                 'dummy': 'sherpa.plot.backends.BasicBackend',
+                 }
```

Comment:
So, this gives me the ability to find the backend object needed to manually change the backend - which I talk about in https://github.com/sherpa/sherpa/pull/1382/files#r877522370 

It would be nice to be able to just create a plot backend without requiring esiting the sherpa code (e.g. if we want to support more plugin-like approaches, a la https://github.com/sherpa/sherpa/pull/1382/files#r878118535 ). I guess a user can always just add entries to this dictionary.

Reply:
You can create new backends any time - This dictionary is only used when parsing the `.sherpa.rc` file. At that point (without using entrypoints or fancy setuptools magic) sherpa only knows about the build-in backends, s it's sufficient to list those here. That's also why these short strings ("dummy") work in the `.sherpa.rc` file, but if you change the backend in the running code you need `sherpa.plot.backend = sherpa.plots.backends.MyFancyBackend()`.

That is from my initial assumption that most people won't money with the backend, especially the XSPEC style users are likely to just use the default backend. So, changing it does not have bells and whistles, but requires a real line of Python. But if you think it might be useful (and I've thought about that, too), then maybe it's time to think harder.

Reply:
OK, more investigation of entry_points: The mechanisms by which one discovers and uses entry points is changing, with a new format introduced in to the standard library in Python 3.10: https://docs.python.org/3/library/importlib.metadata.html
There is an official (by the Python Software Foundation) backport of that as a separate package called `importlib_metadata` to Python 3.6, so we could use that, but it's still a separate package and I suspect that CM would like to avoid that if possible. We could also use the old mechanism, but that's deprecated for 3.10 already. I have no desire to implement it two ways now, so I suggest the following: I'll try to find a way to simplify this for the backends that we provide in sherpa itself (so I always know they exisit and I don't have to discover them). For those, I use a dict similar to what I already do when reading the `.sherpa.rc` file. Any home-made backend (I suspect they will be few except for maybe some experimental stuff that I write myself) can still be used as `sherpa.plot.backend = backend_instance`. Then, I open an issue to move all that (incl. the internal backends) to entry_points, once 3.10 is our minimum required Python version.

Reply:
Antoher way of doing it would be to use a metaclass that registers backends to a registry on import. Maybe that's best... Astropy does that for their `Table.read(format='..') formats. Again, would be replaced with entrypoint eventually....